lib/rsync.coffee

```diff
@@ -22,33 +21,28 @@ exports.createRsyncStream = (src, dest) ->
 				'--delete'
 				'--hard-links'
 
+				'--timeout', ioTimeout # in seconds
+
 				'--compress-level=9'
 				'--one-file-system'
 				'--only-write-batch', pipePath
 				dest, src
 			]
 
-			# Piping to cat causes /dev/stdout to be a pipe instead of a socket which
-			# allows rsync to open() it and write the batch file.
-			# console.error('Invoking rsync:', 'rsync ' + rsyncArgs.join(' '))
-
-			ps = spawn('rsync', rsyncArgs)
-			.on 'error', (error) ->
-				console.error('rsync error', error)
-				ps.stdout.emit('error', error)
-			.on 'exit', (code, signal) ->
-				if code isnt 0
-					ps.stdout.emit('error', new Error("rsync exited. code: #{code} signal: #{signal}"))
-
-			ps.stderr.pipe(process.stderr)
-			ps.stdout.pipe(process.stdout)
+			log('Invoking rsync:', 'rsync ' + rsyncArgs.join(' '))
+			ps = spawn('rsync', rsyncArgs, stdio: 'ignore')
 
 			# Early Node 8 versions have a bug that force a seek upon creation of
 			# a read stream. The workaround is to pass the file descriptor directly.
 			# See: https://github.com/nodejs/node/issues/19240
 			fs.open pipePath, 'r', (err, fd) ->
 				if err
-					cleanup()
-					reject(err)
+					log('Killing rsync due to error...')
+					ps.kill('SIGUSR1')
+					ps.waitAsync().finally ->
+						log('rsync exited')
```

Comment:
Maybe it's worth changing this message to something other than the success message, for debugging?